enhancements/windows-containers/health-management.md

```diff
@@ -0,0 +1,258 @@
+      "name": "string within command field that will be substituted",
+      "jsonPathNodeObject": "value that will be substituted, taken from the provided jsonPath of the Node object"
+    },
+    "powershellVariablesToSubstitute": {
+      "name": "string within command field that will be substituted",
+      "path": "value that will be substituted, taken from the result of the powershell script at this path"
+    }
+  }
+]
+```
+
+The files key contains the path and checksum of files copied to the VM by WMCO. This will be used to validate that
+essential files have not been tampered with.
+```json
+[
+  {
+    "path": "filepath",
+    "checksum": "checksum of file"
+  }
+]
+```
+
+### Node configuration: WMCO responsibilities
+
+1) For a given Machine or BYOH VM, WMCO copies the payload and WMCD binary to the VM.
+2) WMCO runs WMCD with the bootstrap command, this results in the node object being created.
+3) WMCO annotates the Node object with the `desiredVersion` annotation.
+4) The value of this is set to the current WMCO version.
+5) WMCO starts WMCD as a service.
+6) Reconciliation can successfully return at this point.
+
+### Node configuration: WMCD responsibilities
+
+For node configuration WMCD will have two separate responsibilities:
+
+#### Bootstrap command
+
+When run with the bootstrap command WMCD will do the steps required to ensure that Node is created
+for the VM. This functionality is currently implemented within WMCB with the `init-kubelet` command.
+
+#### Running as a Windows service
+
+WMCD will have an informer for both ConfigMap and Node events. When it is detected that either the desired version
+has been set for the Node, or the ConfigMap specified by the desired version annotation has been created, WMCD will
+configure the Windows services to the specifications listed in relevant the ConfigMap. If an error occurs, an event
+will be generated against the Node object.
+
+When configuring a service, WMCD will ensure that the service exists as specified in the ConfigMap. In order to
+properly do this, it must evaluate the variables listed within the ConfigMap. This will be done using the Node object,
+which can be achieved by searching for a Node with an internal IP equivalent to the IP of the VM, and the powershell
+scripts copied to the VM by WMCO.
+
+On a separate thread, WMCD will be periodically polling the state of Windows services. If there is any change in the
+expected state, the services will be created or modified in order to restore expected state. As multiple threads can
+be modifying Windows service state, the appropriate thread safety procedures must be followed.
+
+### Node de-configuration: WMCO responsibilies
+
+When a user indicates to WMCO that it wishes to remove a BYOH node from the cluster, the following procedure will be
+kicked off.
+
+1) WMCO stops the WMCD service on the VM. The dependent services will be automatically stopped as well.
+2) WMCO runs the deconfigure command of WMCD.
```

Comment:
How will WMCO know when to run the deconfigure command? Is it going to wait for all services to stop? Or just the daemon?